app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -829,5 +703,217 @@ class BrowserTabFragment : Fragment(), FindListener {
         }
     }
 
+    class BrowserTabFragmentRenderer(private val browserTabFragment: BrowserTabFragment) {
+
+        private var lastSeenViewState: ViewState? = null
+
+        fun render(viewState: ViewState) {
```

Comment:
Can we remove this global render method? Now that we have individual view models we could have listeners on those that trigger the relevant render methods directly.

Reply:
This was a big change footprint to make it happen, but it's done. Should be a further improvement on performance; good suggestion!